src/remark/code-block-snippets.js

```diff
@@ -0,0 +1,125 @@
+// @ts-check
+const fs = require('fs-extra');
+const path = require("path");
+const visit = require('unist-util-visit');
+const viewLinkUrl = require('../editUrl.js').viewLinkUrl;
+
+module.exports = function plugin() {
+    let filePath = ""
+
+    /**
+     * Searches for ![code](source) pattern and replaces it with the code block containing the
+     * content of the source file. It also supports region names inside the source file defined
+     * by #region and #endregion literals.
+     *
+     * Examples:
+     * ![code](../examples/my-example/shell-dump.txt)
+     * ![code go](../examples/my-example/src/main.go)
+     * ![code go](../examples/my-example/src/server.go "src/server.go")
+     * ![code go](../examples/my-example/src/server.go#L3)
+     * ![code go](../examples/my-example/src/server.go#L5-L11 "src/server.go")
+     * ![code go](../examples/my-example/src/account.go#native-account "Structure of the Native account")
+     */
+    function visitor(node) {
```

Comment:
We can have some intellisense and type warnings with

```suggestion
     *
     * @typedef {import('mdast').Image} ImageNode
     * @typedef {import('mdast').Code} CodeNode
     * @param {Readonly<Omit<ImageNode, 'type'>> & CodeNode} node
     */
    function visitor(node) {
```